packages/create-liveblocks-app/bin/templates/upgrade/upgrade-template.ts

```diff
@@ -25,43 +25,86 @@ export async function create() {
 
   // === Collect Liveblocks dependencies ==================================================
 
-  const depsToUpgrade = Array.from(
+  const liveblocksDeps = Array.from(
     new Set([
       ...findLiveblocksDependencies(pkg.dependencies),
       ...findLiveblocksDependencies(pkg.devDependencies),
     ])
   );
 
-  if (depsToUpgrade.length === 0) {
+  if (liveblocksDeps.length === 0) {
     console.log(c.bold.redBright("  No Liveblocks packages found"));
     return;
   }
 
-  const latestDependencies = depsToUpgrade.map((dep) => `${dep}@latest`);
+  // Decide which dependencies to upgrade or uninstall
+  const depsToUpgrade = liveblocksDeps.map((d) => {
+    if (d === "@liveblocks/react-comments") {
+      return "@liveblocks/react-ui";
+    } else {
+      return d;
+    }
+  });
+
+  const depsToUninstall = liveblocksDeps.filter(
+    (d) => d === "@liveblocks/react-comments"
+  );
 
   // === Upgrade collected dependencies to latest ===========================================
 
   const pkgManager = await detect();
 
-  let installCommand;
+  let installCmd;
+  let uninstallCmd;
 
   switch (pkgManager) {
     case "yarn":
     case "bun":
     case "pnpm":
-      installCommand = "add";
+      installCmd = "add";
+      uninstallCmd = "remove";
       break;
     case "npm":
     default:
-      installCommand = "install";
+      installCmd = "install";
+      uninstallCmd = "uninstall";
       break;
   }
 
   console.log();
   console.log(c.bold.magentaBright(`✨ Upgrading all Liveblocks packages`));
   console.log();
 
-  await execa(pkgManager, [installCommand, ...latestDependencies], {
-    stdio: "inherit",
-  });
+  if (depsToUninstall.length > 0) {
+    await execa(pkgManager, [uninstallCmd, ...depsToUninstall], {
+      stdio: "inherit",
+    });
+  }
+
+  if (depsToUpgrade.length > 0) {
+    await execa(
+      pkgManager,
+      [installCmd, ...depsToUpgrade.map((dep) => `${dep}@latest`)],
+      { stdio: "inherit" }
+    );
+  }
+
+  // Warn user if dependency on @liveblocks/core is detected
+  if (liveblocksDeps.includes("@liveblocks/core")) {
+    console.log();
+    console.log(
+      c.bold.yellowBright(`⚠️ We detected a dependency on @liveblocks/core.`)
+    );
+    console.log(
+      c.bold.yellowBright(
+        `Please note that @liveblocks/core contains private APIs only.`
+      )
+    );
+    console.log(
+      c.bold.yellowBright(
+        `It is recommended to only rely on @liveblocks/client.`
+      )
+    );
+    console.log();
+  }
```

Comment:
@marcbouchenoire Do you think this is the right coloring style?

Reply:
Not 100% sure but it should be alright 👍